nexus/src/external_api/params.rs

```diff
@@ -182,6 +182,15 @@ pub struct DiskCreate {
     pub size: ByteCount,
 }
 
+/**
+ * Parameters for the [`Disk`] to be attached or detached to an instance
+ */
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct DiskReference {
+    pub disk: Name,
```

Comment:
I think this would be better called `name` since the struct makes clear that it's about a disk. Also "Reference" is kind of loaded in Rust, how about `DiskIdentifier`? I think that's more in line with the direction we're going to end up going re: #485

Reply:
I'm not sure we'd want the key of the param to be `name`. The reason being after we solve for #485 it could technically be either a `Name` or a `UUID` but the key would be the same regardless. At least, that's how I think it would work. If we changed the key to `name` it'd have to also potentially accept `id` which seems like a harder thing to model via the JSON schema. 

I will update the name of the struct though, I agree it's overloaded.

Reply:
The way I picture it looking is we'd pick the name or ID out of the params and then do the lookups to convert it to a proper `DiskIdentifier`, which would have `(org_id, project_id, disk_id)` (and an `InstanceIdentifier` that would look similar), and then we'd pass both entire structs to the nexus function:

```rs
nexus.instance_disks_attach(instance_identifier, disk_identifier)
```

So IMO that looks different enough from what we have now that I think it's all right to call it `name` for short-term clarity.